lib/logstash/filters/geoip/database_manager.rb

```diff
@@ -0,0 +1,265 @@
+require "logstash/util/loggable"
+require "logstash/agent"
+require "csv"
+require "digest"
+require "faraday"
+require "json"
+require "zlib"
+require "stud/try"
+require "down"
+require "rufus/scheduler"
+
+# The mission of DatabaseManager is to ensure the plugin running an up-to-date MaxMind database and thus users are compliant with EULA.
+# DM does a daily checking by calling an endpoint to notice a version update.
+# It records the update timestamp and md5 of the database in the metadata file to keep track of versions and the number of days disconnects to the endpoint.
+# Once a new database version release, DM downloads it, and GeoIP Filter uses it on-the-fly.
+# If the last update timestamp is 25 days ago, a warning message shows in the log; if it is 30 days ago, the GeoIP Filter should shutdown in order to be compliant.
+# There are online mode and offline mode in DM. `online` is for automatic database update while `offline` is for static database path provided by users or Logstash running in <= 7.11
+module LogStash module Filters module Geoip class DatabaseManager
+  include LogStash::Util::Loggable
+
+  def initialize(geoip, database_path, database_type)
+    @geoip = geoip
+    @mode = get_mode(database_path)
+    @metadata_path = get_file_path("metadata.csv")
+    @database_type = database_type
+    @database_path = patch_database_path(database_path)
+
+    if @mode == :online
+      raise "logstash-filter-geoip is under elastic Basic License. You are running in open source version, hence the pipeline stops" if LogStash::OSS
```

Comment:
We haven't decided yet to block OSS user completely from running this download manager